module/apps/actor-importer.js

```diff
@@ -1,306 +1,528 @@
+          i++
+        } else if (sections[i].match(this.getRegEx('(' + this.keys.sectionLangauges + ')', 'i')) !== null && typeof sections[i + 1] !== 'undefined') {
+          // If section is languages
+          this.text = ('\n' + this.text + '\n').replace(sections[i], '\n').trim()
+          this.processSkills(sections[i + 1], 'languages')
+          i++
+        } else if (sections[i].match(this.getRegEx('(' + this.keys.sectionSpells + ')', 'i')) !== null && typeof sections[i + 1] !== 'undefined') {
+          // If section is spells
+          this.text = ('\n' + this.text + '\n').replace(sections[i], '\n').trim()
+          this.processSpells(sections[i + 1])
+          i++
+        }
+      }
     }
-    return 'npc'
+    // Any remaining text add to GM notes so you can easily see if there are any obvious issue or just general notes
+    this.parsed.gmnotes = this.toHTML(this.text)
+    return this.parsed
   }
 
   /**
-   * Create an entity (`npc`or `creature`) from the object with the already parsed entity data
-   * @param {Object} pc object with the data extracted from the character as returned from `parseCharacter`
-   * @param {String} entityTypeString entity type obtained from the user input, will be sanitized by calling `entityType`
+   * Create an entity (`npc` or `creature`) from the object with the already parsed entity data
+   * @param {Object} characterData object with the data extracted from the character
+   * @param {String} entityTypeString entity type obtained from the user input
    * @returns {Actor} the created foundry `Actor`
    */
-  async createEntity (pc, entityTypeString) {
+  async createEntity (characterData, entityType) {
     const importedCharactersFolder = await this.createImportCharactersFolderIfNotExists()
-
-    const npc = await Actor.create({
-      name: pc.name,
-      type: this.entityType(entityTypeString),
+    if (entityType !== 'npc') {
+      entityType = 'creature'
+    }
+    const actorData = {
+      name: characterData.name,
+      type: entityType,
       folder: importedCharactersFolder.id,
-      data: {}
-    })
-    // debugger
-    // Add the skills
-    await this.addTheSkills(pc, npc)
-    // Add the languages
-    await this.addTheLanguages(pc, npc)
-    // Add the spells
-    await this.addTheSpells(pc, npc)
-    // Handle the attacks
-    await this.handleTheAttacks(pc, npc)
-    await this.updateActorData(pc, npc)
-
+      data: characterData.actor
+    }
+    const npc = await Actor.create(actorData)
+    await npc.createEmbeddedDocuments('Item', characterData.items, { renderSheet: false })
+    const updateData = {}
+    let value = 0
+    if (typeof characterData.actor.attribs.hp?.value !== 'undefined') {
+      value = Math.max(0, Number(characterData.actor.attribs.hp.value))
+      if (value !== Number(npc.hpMax)) {
+        updateData['data.attribs.hp.auto'] = false
+        updateData['data.attribs.hp.value'] = value
+        updateData['data.attribs.hp.max'] = value
```

Comment:
this 3 `updateData` lines could be extracted to a function to simplify all this ifs

Reply:
The numeric values now call a function to change updateData